service/src/main/java/bio/terra/tanagra/service/authentication/JwtUtils.java

```diff
@@ -71,6 +74,18 @@ public static UserId verifyJwtAndGetUserid(
     }
   }
 
+  // Check and cast the object
+  @SuppressWarnings("unchecked")
+  private static <T> T castObject(String objName, @Nullable Object obj, Class<T> clazz) {
+    if ((obj != null) && !clazz.equals(obj.getClass())) {
+      throw new IllegalArgumentException(
```

Comment:
Since this is just throwing an exception, instead of suppressing and using null, I think you can just catch a `ClassCastException` in the catch block, no need for this helper method. e.g.
```
      return makeUserId(
         (String) payloadMap.getOrDefault(CLAIM_SUB, null),
          (String) payloadMap.getOrDefault(CLAIM_EMAIL, null),
          jwt);

    } catch (ClassCastException e) {
      LOGGER.error("Subject id or email claims are not strings as expected: {}, {}", payloadMap.getOrDefault(CLAIM_SUB, null), payloadMap.getOrDefault(CLAIM_EMAIL, null)));
      throw new InvalidCredentialsException(...)
```

Reply:
Forgot about ClassCastException, its cleaner.

Reply:
Skipped an explicit log here - since the other catch both will still log the exception cause.